File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] nums) {
        Arrays.sort(nums, (a, b) -> Integer.compare(a[0], b[0]));
        
        List<int[]> list= new ArrayList<>(); 

        int[] curr = nums[0]; // [1,3]

        for (int i = 0; i < nums.length; i++) {
            if (curr[1] >= nums[i][0]) {
                // merge
                curr[1] = Math.max(nums[i][1], curr[1]); // new interval
            } else {// no merge, move curr
            list.add(curr); 
                curr = nums[i];
            }
        }
         
                 list.add(curr);

        
        return list.toArray(new int[list.size()][]);


    }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        intervals.sort(key=lambda x:x[0]) # n*log n 

        curr=intervals[0]
        res=[]
        
        for i in range(1,len(intervals)): 
            
            if curr[1]>=intervals[i][0] : 
                #merge 
                curr[1]=max(curr[1],intervals[i][1]) # 1,6
                
            else: 
                res.append(curr)
                curr= intervals[i]
        res.append(curr)
        return res



        